### ISSUE TITLE
TypeError: Cannot read property 'exports' of undefined

### ISSUE THREAD

**Issue by designcourse** [clbond/angular-ssr#35]
I'm following the instructions for setting up the most simple example based on your instructions.

I generated a new project with:
ng new projname --routing

I generated some components to test out the SEO stuff, linked to them with routerLink, set them up with paths in the app-routing.module.ts, and did your steps:

ng build
.\node_modules\.bin\ng-render

It outputs:
[info] Rendering application from source (working path: C:\Users\gary\AppData\Local\Temp\eed79cbf415c8)

...and then in about 60-120 seconds, outputs this:

```
C:\Users\gary\code\ang-seo-pre\node_modules\angular-ssr\source\application\static\find.ts:16
  const exports: Map<string, any> = sourceFile['symbol'].exports;
                                                        ^
TypeError: Cannot read property 'exports' of undefined
    at Object.exports.exportClause (C:\Users\gary\code\ang-seo-pre\node_modules\angular-ssr\source\application\static\find.ts:16:57)
    at _loop_1 (C:\Users\gary\code\ang-seo-pre\node_modules\angular-ssr\source\application\static\modules\root.ts:60:29)
    at Object.exports.discoverRootModule (C:\Users\gary\code\ang-seo-pre\node_modules\angular-ssr\build\application\static\modules\root.js:59:9)
    at Object.exports.loadApplicationModule (C:\Users\gary\code\ang-seo-pre\node_modules\angular-ssr\source\application\compiler\options.ts:57:14)
    at WebpackCompiler.compile (C:\Users\gary\code\ang-seo-pre\node_modules\angular-ssr\source\application\compiler\webpack\compiler.ts:30:38)
    at ApplicationBuilderFromSource.build (C:\Users\gary\code\ang-seo-pre\node_modules\angular-ssr\source\application\builder\from-source.ts:18:29)
    at Object.<anonymous> (C:\Users\gary\code\ang-seo-pre\node_modules\angular-ssr\source\bin\render.ts:36:29)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\gary\code\ang-seo-pre\node_modules\angular-ssr\bin\render.js:3:1)
    at Module._compile (module.js:570:32)
```

I'm sure I'm just something doing stupid or overlooked a step?  Let me know, thanks!

**Comment by Florian007** [comment]
Thank you for this great tool. But if I create a new own project like designcourse i get the exact same error. Only if I clone your rep, change into the "examples/cli" folder, make "npm i" and then "ng build" followed by "npm run ng-render --environment=prod --preboot=true --debug --inline" it works as expected. I compared, copied and tested the package.json, angular-cli.json and tsconfig.json to my previously self created new project with the cli but I always got the same error mentioned above. I don't know what's the differnce and what to change to work in my project. Help would be highly appreciated. Thanks in advance.

**Comment by Florian007** [comment]
I managed to get rid of the error with replacing the content of "src/tsconfig.app.json" with the content from the original angular-ssr repository file "examples/cli/src/tsconfig.app.json" :

{
  "compilerOptions": {
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "lib": [
      "es2016",
      "dom"
    ],
    "outDir": "../out-tsc/app",
    "module": "es2015",
    "baseUrl": "",
    "types": []
  },
  "exclude": [
    "test.ts",
    "**/*.spec.ts"
  ]
}

Additonally I think "preboot" and "hammerjs" should be installed
npm i hammerjs preboot --save

and in main.ts should following code be added:
import {prebootClient} from 'preboot/__build/src/browser/preboot_browser';
import 'hammerjs';

and the bootsrap line replaced with: 
platformBrowserDynamic().bootstrapModule(AppModule).then(() => prebootClient().complete());

**Comment by clbond** [comment]
Hmm! There is indeed something going haywire with new `ng new` projects. I am not sure why yet. Looking into it

**Comment by clbond** [comment]
This is resolved in `0.1.105`:

```
Bond-MacBook:~ bond$ ng new projname --routing
installing ng
  create .editorconfig
  create README.md
  create src/app/app-routing.module.ts
  create src/app/app.component.css
  create src/app/app.component.html
  create src/app/app.component.spec.ts
  create src/app/app.component.ts
  create src/app/app.module.ts
  create src/assets/.gitkeep
  create src/environments/environment.prod.ts
  create src/environments/environment.ts
  create src/favicon.ico
  create src/index.html
  create src/main.ts
  create src/polyfills.ts
  create src/styles.css
  create src/test.ts
  create src/tsconfig.app.json
  create src/tsconfig.spec.json
  create src/typings.d.ts
  create .angular-cli.json
  create e2e/app.e2e-spec.ts
  create e2e/app.po.ts
  create e2e/tsconfig.e2e.json
  create .gitignore
  create karma.conf.js
  create package.json
  create protractor.conf.js
  create tsconfig.json
  create tslint.json
Successfully initialized git.
Installing packages for tooling via npm.
cd ^C
Bond-MacBook:~ bond$ cd projname/
Bond-MacBook:projname bond$ yarn
Bond-MacBook:projname bond$ yarn && yarn add angular-ssr@0.1.105 && npm run build && ./node_modules/.bin/ng-render
yarn install v0.23.2
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Saved lockfile.
✨  Done in 23.88s.
yarn add v0.23.2
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "angular-ssr@0.1.104" has unmet peer dependency "@angular/service-worker@>=1.0.0 || >=1.0.0-beta.8".
[4/4] 📃  Building fresh packages...
success Saved lockfile.
success Saved 218 new dependencies.
├─ @types/chalk@0.4.31
├─ @types/node-fetch@1.6.7
├─ @types/node@6.0.73
├─ @types/source-map@0.5.0
├─ @types/tapable@0.2.3
├─ @types/uglify-js@2.6.28
├─ @types/webpack@2.2.15
├─ abbrev@1.1.0
├─ acorn@4.0.11
├─ angular-ssr@0.1.104
├─ ansi-align@2.0.0
├─ ansi-regex@2.1.1
├─ ansicolors@0.3.2
├─ ansistyles@0.1.3
├─ aproba@1.1.1
├─ archy@1.0.0
├─ array-filter@0.0.1
├─ array-map@0.0.0
├─ array-reduce@0.0.0
├─ asap@2.0.5
├─ bluebird@3.5.0
├─ boxen@1.1.0
├─ builtins@1.0.3
├─ call-limit@1.1.0
├─ capture-stack-trace@1.0.0
├─ chownr@1.0.1
├─ cli-boxes@1.0.0
├─ cmd-shim@2.0.2
├─ columnify@1.5.4
├─ commander@2.9.0
├─ concat-stream@1.6.0
├─ config-chain@1.1.11
├─ configstore@3.1.0
├─ copy-concurrently@1.0.3
├─ create-error-class@3.0.2
├─ cross-spawn-async@2.2.5
├─ crypto-random-string@1.0.0
├─ cyclist@0.2.2
├─ debuglog@1.0.1
├─ defaults@1.0.3
├─ define-properties@1.1.2
├─ dezalgo@1.0.3
├─ domino@1.0.43
├─ dot-prop@4.1.1
├─ duplexer@0.1.1
├─ duplexer3@0.1.4
├─ duplexify@3.5.0
├─ editor@1.0.0
├─ encoding@0.1.12
├─ end-of-stream@1.4.0
├─ es-abstract@1.7.0
├─ es-to-primitive@1.1.1
├─ event-stream@3.3.4
├─ execa@0.4.0
├─ flush-write-stream@1.0.2
├─ foreach@2.0.5
├─ from@0.1.7
├─ from2@1.3.0
├─ fs-vacuum@1.2.10
├─ fs-write-stream-atomic@1.0.10
├─ fstream-ignore@1.0.5
├─ fstream-npm@1.2.1
├─ fstream@1.0.11
├─ function-bind@1.1.0
├─ gauge@2.7.4
├─ get-stream@3.0.0
├─ got@6.7.1
├─ graceful-fs@4.1.11
├─ has-unicode@2.0.1
├─ hosted-git-info@2.4.2
├─ iconv-lite@0.4.15
├─ iferr@0.1.5
├─ imurmurhash@0.1.4
├─ inflight@1.0.6
├─ inherits@2.0.3
├─ ini@1.3.4
├─ init-package-json@1.10.1
├─ is-callable@1.1.3
├─ is-date-object@1.0.1
├─ is-npm@1.0.0
├─ is-obj@1.0.1
├─ is-redirect@1.0.0
├─ is-regex@1.0.4
├─ is-retry-allowed@1.1.0
├─ is-stream@1.1.0
├─ is-symbol@1.0.1
├─ jju@1.3.0
├─ json-parse-helpfulerror@1.0.3
├─ jsonparse@1.3.1
├─ jsonschema@1.1.1
├─ JSONStream@1.3.1
├─ karma-sourcemap-loader@0.3.7
├─ karma-webpack@2.0.3
├─ latest-version@3.1.0
├─ lazy-property@1.0.0
├─ lazy-req@2.0.0
├─ lockfile@1.0.3
├─ lodash._baseindexof@3.1.0
├─ lodash._baseuniq@4.6.0
├─ lodash._bindcallback@3.0.1
├─ lodash._cacheindexof@3.0.2
├─ lodash._createcache@3.1.2
├─ lodash._createset@4.0.3
├─ lodash._getnative@3.9.1
├─ lodash._root@3.0.1
├─ lodash.clonedeep@4.5.0
├─ lodash.restparam@3.6.1
├─ lodash.union@4.6.0
├─ lodash.uniq@4.5.0
├─ lodash.without@4.4.0
├─ lowercase-keys@1.0.0
├─ lru_map@0.3.3
├─ lru-cache@4.0.2
├─ make-dir@1.0.0
├─ map-stream@0.1.0
├─ mississippi@1.3.0
├─ mkdir-recursive@0.3.0
├─ mkpath@1.0.0
├─ move-concurrently@1.0.1
├─ mute-stream@0.0.7
├─ node-fetch@1.6.3
├─ node-gyp@3.6.1
├─ nopt@4.0.1
├─ normalize-git-url@3.0.2
├─ normalize-package-data@2.3.8
├─ npm-cache-filename@1.0.2
├─ npm-install-checks@3.0.0
├─ npm-package-arg@4.2.1
├─ npm-registry-client@8.1.1
├─ npm-run-all@4.0.2
├─ npm-run-path@1.0.0
├─ npm-user-validate@0.1.5
├─ npm@4.6.1
├─ npmlog@4.1.0
├─ object-keys@1.0.11
├─ once@1.4.0
├─ opener@1.4.3
├─ osenv@0.1.4
├─ package-json@4.0.1
├─ parallel-transform@1.1.0
├─ path-is-inside@1.0.2
├─ path-key@1.0.0
├─ pause-stream@0.0.11
├─ preboot@4.5.2
├─ prepend-http@1.0.4
├─ promzard@0.3.0
├─ proto-list@1.2.4
├─ ps-tree@1.1.0
├─ pump@1.0.2
├─ pumpify@1.3.5
├─ rc@1.2.1
├─ read-cmd-shim@1.0.1
├─ read-installed@4.0.3
├─ read-package-json@2.0.5
├─ read-package-tree@5.1.5
├─ read@1.0.7
├─ readable-stream@2.2.9
├─ readdir-scoped-modules@1.0.2
├─ realize-package-specifier@3.0.3
├─ registry-auth-token@3.3.1
├─ registry-url@3.1.0
├─ request@2.81.0
├─ retry@0.10.1
├─ rimraf@2.6.1
├─ run-queue@1.0.3
├─ scoped-logger@0.0.19
├─ semver-diff@2.1.0
├─ semver@5.3.0
├─ sha@2.0.1
├─ shebang-command@1.2.0
├─ shebang-regex@1.0.0
├─ shell-quote@1.6.1
├─ slide@1.1.6
├─ sorted-object@2.0.1
├─ sorted-union-stream@2.1.3
├─ source-map-support@0.4.15
├─ split@0.3.3
├─ stream-combiner@0.0.4
├─ stream-each@1.2.0
├─ stream-iterate@1.2.0
├─ stream-shift@1.0.0
├─ string.prototype.padend@3.0.0
├─ strip-ansi@3.0.1
├─ strip-eof@1.0.0
├─ symbol-observable@1.0.4
├─ tar@2.2.1
├─ term-size@0.1.1
├─ text-table@0.2.0
├─ through@2.3.8
├─ through2@2.0.3
├─ timed-out@4.0.1
├─ typedarray@0.0.6
├─ typescript@2.3.2
├─ uglify-js@2.8.26
├─ uid-number@0.0.6
├─ umask@1.1.0
├─ unique-filename@1.1.0
├─ unique-slug@2.0.0
├─ unique-string@1.0.0
├─ unzip-response@2.0.1
├─ update-notifier@2.1.0
├─ url-parse-lax@1.0.0
├─ util-extend@1.0.3
├─ uuid@3.0.1
├─ validate-npm-package-license@3.0.1
├─ validate-npm-package-name@3.0.0
├─ watchpack@1.3.1
├─ wcwidth@1.0.1
├─ webpack-dev-middleware@1.10.2
├─ webpack-dev-server@2.4.5
├─ webpack-sources@0.1.5
├─ which@1.2.14
├─ widest-line@1.0.0
├─ wrappy@1.0.2
├─ write-file-atomic@1.3.4
├─ xdg-basedir@3.0.0
├─ xhr2@0.1.4
└─ xtend@4.0.1
✨  Done in 13.00s.

> projname@0.0.0 build /Users/bond/projname
> ng build

Hash: 1976071a450bd49dd74c                                                              
Time: 13480ms
chunk    {0} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 157 kB {4} [initial] [rendered]
chunk    {1} main.bundle.js, main.bundle.js.map (main) 6.48 kB {3} [initial] [rendered]
chunk    {2} styles.bundle.js, styles.bundle.js.map (styles) 10.5 kB {4} [initial] [rendered]
chunk    {3} vendor.bundle.js, vendor.bundle.js.map (vendor) 2.09 MB [initial] [rendered]
chunk    {4} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]

[info] Rendering application from source (working path: /var/folders/k9/ww5nrgyx687gpx7s_86tf37m0000gn/T/e1a434a8c1568) 
Angular is running in the development mode. Call enableProdMode() to enable the production mode.
Angular is running in the development mode. Call enableProdMode() to enable the production mode.
[info] Rendered route / to /Users/bond/projname/dist/index.html 
```

Thank you for your bug report!
